File: App/BTServer/Care4Toolbox/src/main/java/com/care4u/toolbox/tag/TagDto.java
package com.care4u.toolbox.tag;

import com.care4u.toolbox.ToolboxDto;
import com.care4u.toolbox.sheet.rental.rental_tool.RentalToolDto;
import com.care4u.toolbox.tool.ToolDto;

import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public class TagDto {
	
	private long id;
	
	private String macaddress;
	
	private ToolboxDto toolboxDto;
	
	private ToolDto toolDto;
	
	private RentalToolDto rentalToolDto;
	
	@Builder
	public TagDto(long id, String macaddress, ToolboxDto toolboxDto, ToolDto toolDto, RentalToolDto rentalToolDto) {
		this.id = id;
		this.macaddress = macaddress;
		this.toolboxDto = toolboxDto;
		this.toolDto = toolDto;
		this.rentalToolDto = rentalToolDto;
	}
	
	public TagDto(Tag tag) {
		this.id = tag.getId();
		this.macaddress = tag.getMacaddress();
		this.toolboxDto = new ToolboxDto(tag.getToolbox());
		this.toolDto = new ToolDto(tag.getTool());
		this.rentalToolDto = new RentalToolDto(tag.getRentalTool());
	}
	
}

File: App/BTClient/app/src/main/java/com/mrsmart/standard/tag/TagDto.kt
package com.mrsmart.standard.tag

import com.mrsmart.standard.sheet.rental.RentalToolDto
import com.mrsmart.standard.tool.ToolDto
import com.mrsmart.standard.toolbox.ToolboxDto

data class TagDto(
    val id: Long,
    val macaddress: String,
    val toolboxDto: ToolboxDto,
    val toolDto: ToolDto,
    val rentalToolDto: RentalToolDto,
    val tagGroup: String
)
